sky/backends/backend_utils.py

```diff
@@ -778,6 +778,13 @@ def write_cluster_config(
                 'cluster_name': cluster_name,
                 'num_nodes': num_nodes,
                 'disk_size': to_provision.disk_size,
+                # If the current code is run by controller, propagate the real
+                # calling user which should've been passed in as the
+                # SKYPILOT_USER env var (see spot-controller.yaml.j2).
+                'user': os.environ.get('SKYPILOT_USER',
+                                       os.environ.get('USER', None)),
```

Comment:
nit: In other places, we use `getpass.getuser()`, can we be consistent here, instead of using the `USER` environment variable?

Reply:
Good catch; done.